src/database/PostgreSQL.ts

```diff
@@ -7,44 +7,35 @@ import {SerializedGame} from '../SerializedGame';
 import {Pool, ClientConfig, QueryResult} from 'pg';
 
 export class PostgreSQL implements IDatabase {
-  private client: Pool;
+  protected client: Pool;
 
-  constructor() {
-    const config: ClientConfig = {
+  constructor(
+    config: ClientConfig = {
       connectionString: process.env.POSTGRES_HOST,
-    };
+    }) {
     if (config.connectionString !== undefined && config.connectionString.startsWith('postgres')) {
       config.ssl = {
         // heroku uses self-signed certificates
         rejectUnauthorized: false,
       };
     }
     this.client = new Pool(config);
-    this.client.query('CREATE TABLE IF NOT EXISTS games(game_id varchar, players integer, save_id integer, game text, status text default \'running\', created_time timestamp default now(), PRIMARY KEY (game_id, save_id))', (err) => {
-      if (err) {
-        throw err;
-      }
-    });
-    this.client.query('CREATE TABLE IF NOT EXISTS game_results(game_id varchar not null, seed_game_id varchar, players integer, generations integer, game_options text, scores text, PRIMARY KEY (game_id))', (err) => {
-      if (err) {
-        throw err;
-      }
-    });
-
-    this.client.query('CREATE INDEX IF NOT EXISTS games_i1 on games(save_id)', (err) => {
-      if (err) {
-        throw err;
-      }
-    });
-    this.client.query('CREATE INDEX IF NOT EXISTS games_i2 on games(created_time )', (err) => {
-      if (err) {
-        throw err;
-      }
-    });
   }
 
-  async initialize(): Promise<void> {
-
+  public async initialize(): Promise<void> {
+    return this.client.query('CREATE TABLE IF NOT EXISTS games(game_id varchar, players integer, save_id integer, game text, status text default \'running\', created_time timestamp default now(), PRIMARY KEY (game_id, save_id))')
```

Comment:
```suggestion
    await this.client.query('CREATE TABLE IF NOT EXISTS games(game_id varchar, players integer, save_id integer, game text, status text default \'running\', created_time timestamp default now(), PRIMARY KEY (game_id, save_id))')
```

With this now inside an `async` function making use of `await` will spare us from the `then` ladders of yore.

Reply:
Yes, this makes sense. Should this also remove all the `then` calls and just call await four times in a row?